archive/schema.py

```diff
@@ -0,0 +1,176 @@
+        schema = super().get_schema(*args, **kwargs)
+        schema['info']['title'] = 'Science Archive API'
+        schema['info']['description'] = 'API documentation for the OCS Science Archive'
+        schema['info']['version'] = get_version(version_scheme=version_scheme, local_scheme='no-local-version')
+        return schema
+
+
+class ScienceArchiveSchema(AutoSchema):
+    def __init__(self, tags=None, operation_id_base=None, component_name=None):
+        super().__init__(tags=tags, operation_id_base=operation_id_base, component_name=component_name)
+
+    def get_operation_id(self, path, method):
+        """
+        This method is used to determine the descriptive name of the endpoint displayed in the documentation.
+        Allow this to be overridden in the view - a view that defines get_endpoint_name can override the default
+        DRF naming scheme.
+        """
+        operation_id = super().get_operation_id(path, method)
+        if getattr(self.view, 'get_endpoint_name', None) is not None:
+            name_override = self.view.get_endpoint_name()
+            # For some viewsets, we may not have specified a name for an action - guard against that
+            if name_override is not None:
+                operation_id = name_override
+
+        return operation_id
+
+    def get_operation(self, path, method):
+        """
+        This method is used to determine, among other things, the request and response bodies for a particular endpoint.
+        We override this to support specifying our own request and response bodies. Any view that implements get_example_response
+        and/or get_example_request can provide their own custom request and response bodies to display in the OpenAPI docs.
+        """
+        operations =  super().get_operation(path, method)
+        # If the view has implemented get_example_response, then use it to present in the documentation
+        if getattr(self.view, 'get_example_response', None) is not None:
+            example_response = self.view.get_example_response()
+            # For viewsets, not all actions may have example responses defined, so this method may return None
+            if example_response is not None:
+                status_code = example_response.status_code
+                example_data = example_response.data
+                content_type = example_response.content_type if example_response.content_type is not None else 'application/json'
+                operations['responses'] = {status_code: {'content': {content_type: {'example': example_data}}}}
+
+        # If the view has implemented get_example_request, then use it to present in the documentation
+        if getattr(self.view, 'get_example_request', None) is not None:
+            example_request = self.view.get_example_request()
+            if example_request is not None:
+                operations['requestBody']['content']['application/json']['example'] = example_request
+
+        return operations
+
```

Comment:
Again, once this is released in a new DRF release we can remove this stuff. Makes me think the ~60 lines of common code between the obs portal and this project don't necessarily warrant a new common library with these doc overrides, but let me know what you think.

Reply:
I think if the code here is going to be removed soon anyway, some common code here is totally fine